### PR TITLE
feat(user form): add account expiry date field (edit & new)

### PR DESCRIPTION
**Pull request by Mohammer5**
Fixes [DHIS2-10364](https://jira.dhis2.org/browse/DHIS2-10364)

* Adds a date field
* Displays the date field in the user new & edit form

### CODE REVIEW

File: i18n/en.pot [hunk 1]
```diff
@@ -492,6 +492,9 @@ msgstr ""
 msgid "External authentication only (OpenID or LDAP)"
 msgstr ""
 
+msgid "Expire date"
```
Comment (HendrikThePendric) [inline i18n/en.pot:495]: Not sure if this should perhaps be `Expiry date` / `Expiration date`.
Perhaps @cooper-joe could sign off?

File: src/api/utils.js [hunk 2]
```diff
@@ -156,6 +157,10 @@ export const parseUserSaveData = (
         addValueAsProp(cred, values[propName], propName)
     )
 
+    if (!cred[EXPIRE_DATE] && typeof cred[EXPIRE_DATE] !== 'undefined') {
+        delete cred[EXPIRE_DATE]
+    }
```
Comment (HendrikThePendric) [inline src/api/utils.js:162]: It's very good you caught this. Please add a comment above line 160 with a link to the backend bugfix issue.

**Review by HendrikThePendric — APPROVED** [review]
Excellent! Code-wise this is looking good. Perhaps you would want to wait merging until @cooper-joe comments on the text and you've added a comment with link to JIRA issue.

File: i18n/en.pot (hunk 1 above)
Comment (cooper-joe) [reply i18n/en.pot:495]: I would go for `Expiration date` 👍

File: src/api/utils.js (hunk 2 above)
Comment (Mohammer5) [reply src/api/utils.js:163]: Added to the commit

File: i18n/en.pot (hunk 1 above)
Comment (Mohammer5) [reply i18n/en.pot:495]: The field on the user credentials is called `accountExpiry`, so maybe `Account expiration date`?

Comment (cooper-joe) [reply i18n/en.pot:495]: Yes, I think `Account expiration date` makes a lot of sense in that case

**Pull request merged**